pkg/util/system/cgroup_resource.go

```diff
@@ -18,124 +18,290 @@ package system
+	CPUTasks     = DefaultFactory.New(CPUTasksName, CgroupCPUDir)
+	CPUProcs     = DefaultFactory.New(CPUProcsName, CgroupCPUDir)
+
+	CPUSet = DefaultFactory.New(CPUSetCPUSName, CgroupCPUSetDir)
+
+	CPUAcctUsage = DefaultFactory.New(CPUAcctUsageName, CgroupCPUAcctDir)
+
+	MemoryLimit            = DefaultFactory.New(MemoryLimitName, CgroupMemDir)
+	MemoryUsage            = DefaultFactory.New(MemoryUsageName, CgroupMemDir)
+	MemoryStat             = DefaultFactory.New(MemoryStatName, CgroupMemDir)
+	MemoryWmarkRatio       = DefaultFactory.New(MemoryWmarkRatioName, CgroupMemDir).WithValidator(MemoryWmarkRatioValidator).WithCheckSupported(SupportedIfAnolisOS)
+	MemoryWmarkScaleFactor = DefaultFactory.New(MemoryWmarkScaleFactorName, CgroupMemDir).WithValidator(MemoryWmarkScaleFactorFileNameValidator).WithCheckSupported(SupportedIfAnolisOS)
+	MemoryWmarkMinAdj      = DefaultFactory.New(MemoryWmarkMinAdjName, CgroupMemDir).WithValidator(MemoryWmarkMinAdjValidator).WithCheckSupported(SupportedIfAnolisOS)
+	MemoryMin              = DefaultFactory.New(MemoryMinName, CgroupMemDir).WithValidator(NaturalInt64Validator).WithCheckSupported(SupportedIfAnolisOS)
+	MemoryLow              = DefaultFactory.New(MemoryLowName, CgroupMemDir).WithValidator(NaturalInt64Validator).WithCheckSupported(SupportedIfAnolisOS)
+	MemoryHigh             = DefaultFactory.New(MemoryHighName, CgroupMemDir).WithValidator(NaturalInt64Validator).WithCheckSupported(SupportedIfAnolisOS)
+	MemoryPriority         = DefaultFactory.New(MemoryPriorityName, CgroupMemDir).WithValidator(MemoryPriorityValidator).WithCheckSupported(SupportedIfAnolisOS)
+	MemoryUsePriorityOom   = DefaultFactory.New(MemoryUsePriorityOomName, CgroupMemDir).WithValidator(MemoryUsePriorityOomValidator).WithCheckSupported(SupportedIfAnolisOS)
+	MemoryOomGroup         = DefaultFactory.New(MemoryOomGroupName, CgroupMemDir).WithValidator(MemoryOomGroupValidator).WithCheckSupported(SupportedIfAnolisOS)
+
+	BlkioReadIops  = DefaultFactory.New(BlkioTRIopsName, CgroupBlkioDir).WithValidator(NaturalInt64Validator)
+	BlkioReadBps   = DefaultFactory.New(BlkioTRBpsName, CgroupBlkioDir).WithValidator(NaturalInt64Validator)
+	BlkioWriteIops = DefaultFactory.New(BlkioTWIopsName, CgroupBlkioDir).WithValidator(NaturalInt64Validator)
+	BlkioWriteBps  = DefaultFactory.New(BlkioTWBpsName, CgroupBlkioDir).WithValidator(NaturalInt64Validator)
+
+	knownCgroupResources = []Resource{
+		CPUStat,
+		CPUShares,
+		CPUCFSQuota,
+		CPUCFSPeriod,
+		CPUBurst,
+		CPUTasks,
+		CPUBVTWarpNs,
+		CPUSet,
+		CPUAcctUsage,
+		MemoryLimit,
+		MemoryUsage,
+		MemoryStat,
+		MemoryWmarkRatio,
+		MemoryWmarkScaleFactor,
+		MemoryWmarkMinAdj,
+		MemoryMin,
+		MemoryLow,
+		MemoryHigh,
+		MemoryPriority,
+		MemoryUsePriorityOom,
+		MemoryOomGroup,
+		BlkioReadIops,
+		BlkioReadBps,
+		BlkioWriteIops,
+		BlkioWriteBps,
+	}
+
+	CPUQuotaV2    = DefaultFactory.NewV2(CPUCFSQuotaName, CPUMaxName).WithCheckSupported(SupportedIfCgroupsV2)
```

Comment:
WithCheckSupported(SupportedIfCgroupsV2) is not necessary since it's enabled by default

Reply:
done